lib/activerecord-import/adapters/trilogy_adapter.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+module ActiveRecord::Import::TrilogyAdapter
+  include ActiveRecord::Import::ImportSupport
+  include ActiveRecord::Import::OnDuplicateKeyUpdateSupport
```

Comment:
Thanks for taking the initiative to open this PR. I know it's still draft but have a question here. Why do we need a new adaptor implementation here and not reusing `include ActiveRecord::Import::MysqlAdapter` as it's used with `ActiveRecord::Import::Mysql2Adapter`. I can see many shared code below with MysqlAdapter. Could you please clarify?

Reply:
This is mostly me just tinkering around with things. I'm mostly trying to explore how the project is structured, etc.

Reply:
Ok no rush. Thanks for your reply! I asked out of curiosity because I came into the same problem today while upgrading my project to use trilogy adaptor as well. Let me know if I can help. Otherwise, I can at least help with review.